src/common/github/pull-request-test-workflow.ts

```diff
@@ -53,11 +61,36 @@ export class PullRequestTest extends Component {
       steps: [{uses: 'ottofeller/github-actions/npm-run@main', with: {'node-version': 16, command, directory}}],
     })
 
+    const customJob = (steps: JobStep[]): github.workflows.Job => ({
+      runsOn: options.runsOn ?? ['ubuntu-latest'],
+      permissions: {contents: github.workflows.JobPermission.READ},
+      steps,
+    })
+
     workflow.addJobs({
       lint: job('npm run lint'),
       typecheck: job('npm run typecheck'),
       test: job('npm run test'),
     })
+
+    if (options.lighthouse) {
```

Comment:
This works but seems a little weird. The workflow is used by all projects and adding an option that is needed only in a single one looks not SOLID. Maybe we could add some general purpose extension mechanism (like a customJobs array option) and define the details of these custom jobs in the nextjs project.

However I would not object if @gvidon is ok with it. If we have plans on adding any other frontend template the lighthouse audit step can definitely be consider a part of the workflow template.

Reply:
@quesabe agree. I would probably suggest adding new workflow just for the nextjs projects, by extending this one with lighthouse, and also playwright in the future.

Reply:
@bdoof yeah, I think new PR is a good idea.